lib/pgfinder/matching.py

```diff
@@ -456,30 +487,44 @@ def pick_most_likely_structures(
         in the file for completeness.
     """
 
-    def add_most_likely_structure(group):
+    def add_most_likely_structure(group) -> pd.DataFrame:
+        """
+        Determines the most likely structure.
+
+        Parameters
+        ----------
+        group : pd.DataFrame
+            DataFrame of structures.
+
+        Returns
+        -------
+        pd.DataFrame
+            Dataframe of most likely values.
+        """
         # Sort by lowest absolute ppm first, then break ties with structures (short to long)
         group.sort_values(
-            by=["Delta ppm", "Inferred structure"],
+            by=["Delta ppm", columns["inferred"]["structure"]],
             ascending=[True, False],
             key=lambda k: abs(k) if is_numeric_dtype(k) else k,
             inplace=True,
             kind="stable",
         )
         group.reset_index(drop=True, inplace=True)
 
+        # The hard coded 'Delta ppm' and 'Intensity column names aren't yet handled dynamically
```

Comment:
It seems somewhat similar with the "Inferred structure ..." and "Intensity ..." ones which are only partially dynamic. If this is a faff to sort out though, then there is no need to worry about it for now

Reply:
I didn't address these as it would have required making `pgfinder.input` a dictionary so that the `Intensity` column could be referenced and would have required another entry under `pgfinder` to hold the value of `Delta ppm`.

I'll do those in a separate PR though tomorrow as there is no point in leaving the job half-done. Thanks for picking me up on this.